app/main.py

```diff
@@ -1,25 +1,71 @@
 from abc import ABC
+from collections.abc import Callable
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: object, name: str) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: object, owner: object) -> int:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: object, value: int) -> None:
+        if isinstance(value, int):
+            if self.min_amount <= value <= self.max_amount:
+                setattr(instance, self.protected_name, value)
+            else:
+                raise ValueError("You can't slide here :(")
+        else:
+            raise TypeError("Those are not real human parameters")
 
 
 class Visitor:
-   pass
+    def __init__(self, name: str, age: int, weight: int, height: int) -> None:
+        self.name = name
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class SlideLimitationValidator(ABC):
-    pass
+    def __init__(self, age: int, weight: int, height: int) -> None:
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class ChildrenSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(4, 14)
+    height = IntegerRange(80, 120)
+    weight = IntegerRange(20, 50)
 
 
 class AdultSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(14, 60)
+    height = IntegerRange(120, 220)
+    weight = IntegerRange(50, 120)
 
 
 class Slide:
-    pass
+    def __init__(
+            self,
+            name: str,
+            limitation_class: Callable
+    ) -> None:
+        self.name = name
+        self.limitation_class = limitation_class
+
+    def can_access(self, visitor: Visitor) -> bool:
+        try:
+            if self.limitation_class(
+                    visitor.age,
+                    visitor.weight,
+                    visitor.height
+            ):
```

Comment:
The `limitation_class` is expected to be a class, not a callable function. You should instantiate it with the visitor's attributes and then check if the visitor meets the slide's limitations. Consider adding a method in `SlideLimitationValidator` to validate the visitor's attributes.